### PR TITLE
WIP: Refactor using streamz and optionally dask

### PR DESCRIPTION
**Pull request by stikkireddy**
…t variable creation.

### CODE REVIEW

File: databricks_terraformer/cmds/config.py [hunk 1]
```diff
@@ -22,12 +22,22 @@ def git_url_option(f):
...
+
+
 def delete_option(f):
     return click.option('--delete', is_flag=True,
                         help="When fetching and pulling remote state this will delete any items that are managed "
                              "and not retrieved.")(f)
 
 
+def dask_option(f):
+    return click.option('--dask', is_flag=True, default=False,
```
Comment (itaiw) [inline databricks_terraformer/cmds/config.py:37]: Do we need to add a check and fail if the user select --dask, but do not have dask installed?

File: README.md [hunk 2]
```diff
@@ -128,6 +128,7 @@ make; GIT_PYTHON_TRACE=full databricks-terraformer import \
     --artifact-dir tmp \
     --apply \
     --backend-file tmp/backend.tf
+GIT_PYTHON_TRACE=full databricks-terraformer -v DEBUG export --profile field-eng --git-ssh-url git@github.com:stikkireddy/export-example.git -c /Users/Sri.Tikkireddy/PycharmProjects/databricks-terraformer/databricks_terraformer/sdk/test.yaml --dry-run --dask-mode
 ```
```
Comment (itaiw) [inline README.md:133]: Can we add the optional configuration of dask?

File: databricks_terraformer/cmds/config.py [hunk 3]
```diff
@@ -22,12 +22,22 @@ def git_url_option(f):
                         help="This is the github url you wish to use to manage export and import.")(f)
 
 
+def config_file_option(f):
+    return click.option('--config-path', '-c', type=click.Path(exists=True), required=True,
+                        help="This is the path to the config file for the export.")(f)
+
```
Comment (itaiw) [inline databricks_terraformer/cmds/config.py:28]: Is this in the README.md?
Can we run w/o a config file?

File: databricks_terraformer/misc/utils/__init__.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    "databricks_instance_pool",
+    "databricks_instance_profile",
+    "databricks_secret_scope",
+    "databricks_secret",
+    "databricks_secret_acl",
+    "jobs",
+]
+
+
+def handle_block(resource_data, object, block):
```
Comment (itaiw) [inline databricks_terraformer/misc/utils/__init__.py:21]: I have a feeling that we do not need these anymore.
Or will this be part of a big cleanup after this PR?

File: databricks_terraformer/sdk/generators/__init__.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# class HCLDictionaryBuilder:
+#
+#     def __init__(self, src_dictionary):
+#         self.__src_dictionary = src_dictionary
```
Comment (itaiw) [inline databricks_terraformer/sdk/generators/__init__.py:4]: Wh are these all commented out?

File: databricks_terraformer/cmds/config.py (hunk 1 above)
Comment (stikkireddy) [reply databricks_terraformer/cmds/config.py:37]: it will fail because of attempt to import

File: README.md (hunk 2 above)
Comment (stikkireddy) [reply README.md:133]: sorry this was checked in by accident but ya i can add docs for it

File: databricks_terraformer/cmds/config.py (hunk 3 above)
Comment (stikkireddy) [reply databricks_terraformer/cmds/config.py:28]: we cannot run without config file. that is where all the configs come from but the config file should be described in detail.

File: databricks_terraformer/misc/utils/__init__.py (hunk 4 above)
Comment (stikkireddy) [reply databricks_terraformer/misc/utils/__init__.py:21]: big cleanup after pr we may want the code still in the repo to reuse incase we need to copy and paste :-)

File: databricks_terraformer/sdk/generators/__init__.py (hunk 5 above)
Comment (stikkireddy) [reply databricks_terraformer/sdk/generators/__init__.py:4]: wip :D didnt want it being called automatically in the init script sorry

File: databricks_terraformer/cmds/config.py [hunk 6]
```diff
@@ -22,12 +22,22 @@ def git_url_option(f):
                         help="This is the github url you wish to use to manage export and import.")(f)
```
Comment (itaiw) [inline databricks_terraformer/cmds/config.py:23]: I missed it somewhere. I tried to run my version and it failed on:
ModuleNotFoundError: No module named 'yaml'

Can you update dev-requirements.txt to reflect this?

**Pull request closed without merging**